### DISCUSSION TITLE
Easy way to access `CATEGORY_TITLES`/`CATEGORY_DESCRIPTIONS` from `tags.tmpl`?

### DISCUSSION THREAD

**Discussion by kai-tub** [getnikola/nikola#3583] [Q&A] [answered]
Hi :)

Thank you again for providing such a nice static site generator!
I am working on customizing my personal theme and the current goal is to have a _category-based_ landing page.

With `CATEGORY_TITLES`/`CATEGORY_DESCRIPTIONS` I can update the category information into a _prettier_ version.
These variables are then used when the category (index) page itself is rendered.
Now I would like to be able to _re-use_ the information in the `tags.tmpl` page such that I can use the _pretty_ category name as a link with the description to provide extra information.
The category name and description would then be combined into a _card_ with an optional image for easier navigation.

Is there an easy way to accomplish this?
Or am I too stuck on the `tags.tmpl` and should just create a different template and try to use that?

**Comment by kai-tub** [comment]
Ok, a quick hack is to use `GLOBAL_CONTEXT`.
I've added a function that will search through `TITLES` and `DESCRIPTIONS` for the given key and returns the desired information.
It is good enough for now, but if there is an easy way to access the data from `tags.tmpl` as it does for `tag.tmpl` I would still like to know :)

**Comment by Kwpolska** [comment, marked as answer]
I think they should be available as dictionaries under the names of `category_titles` and `category_descriptions`.

**Reply by kai-tub** [reply]
Hmm.
Maybe I am missing some jinja basics, but if I add ` {{ category_titles.get(XXX) }}` to the `tags.tmpl` file I get: ```jinja2.exceptions.UndefinedError: 'category_titles' is undefined```

I have set `category_titles` and can confirm it by looking at the output of `tag.tmpl` (_not_ `tags.tmpl`) and verify that the _pretty_ name is used.
The only place where the dictionary is used in code is in [categories.py](https://github.com/getnikola/nikola/blob/8cc31e0230a7209cc0bca558a9d085b2e179fe8c/nikola/plugins/task/categories.py#L193).
The code _looks_ like it is mainly used to produce the tag/category pages and not the categories page, but I am not sure...

**Reply by Kwpolska** [reply]
Right, that’s what was happening. Including `category_titles` etc in `tags.tmpl` makes sense, so I fixed that in PR #3585.

**Reply by kai-tub** [reply]
Thanks!
